src/record/record-transition.js

```diff
@@ -281,12 +346,16 @@ RecordTransition.prototype._flushVersionExists = function () {
  * @private
  * @returns {void}
  */
-RecordTransition.prototype._onCacheResponse = function (error) {
+RecordTransition.prototype._onCacheResponse = function (currentStep, error) {
+  this._cacheResponses--
+  this._writeError = this._writeError || error
   if (error) {
-    this._onFatalError(error)
+    this._onFatalError('cacheError')
```

Comment:
why remove error? we need to log it somehow to figure out if it was due to a connection loss / etc